components/usage/pkg/db/cost_center.go

```diff
@@ -130,12 +132,20 @@ func (c *CostCenterManager) UpdateCostCenter(ctx context.Context, costCenter Cos
 					return CostCenter{}, err
 				}
 			}
+			// we don't manage stripe billing cycle
+			costCenter.NextBillingTime = VarcharTime{}
+		} else if costCenter.BillingStrategy == CostCenter_Other {
+			// cancelled from stripe reset the spending limit
+			if costCenter.ID.IsEntity(AttributionEntity_Team) {
+				costCenter.SpendingLimit = c.cfg.ForTeams
+			} else {
+				costCenter.SpendingLimit = c.cfg.ForUsers
+			}
+			// see you next month
+			costCenter.NextBillingTime = NewVarcharTime(now.AddDate(0, 1, 0))
```

Comment:
I think the individual field assignment into a CostCenter object really hurts readability here. The alternative approach is to construct the CostCenter in each branch seperately (with some possible duplication) but making it very explicit what each branch will result in.